### PR TITLE
CS featured news pod in place of hero image

### PR DESCRIPTION
**Pull request by chrisjpurdy**
This also moves news and page fragments into RTK Query, which made it easier to make this change. 
CS team seem happy with it after showing them some screenshots.

image

### CODE REVIEW

**Review by mlt47 — CHANGES_REQUESTED** [review]
Small change suggestion to the ordering logic.

You should send CS a message on Slack letting them know how to tag a news item as featured, noting that all featured content will appear before other content and that if there are no featured content the first item is selected.

The layout changes on first load as a signed in user were not _too_ bad for me but I do have quick internet.
I like the way it looks on desktop but does look a little strange on smaller devices to my eye but CS are happy so happy to be overruled.

File: src/app/state/slices/api/index.ts [hunk 1]
```diff
@@ -100,6 +100,34 @@ const isaacApi = createApi({
...
+               url: `/pages/pods/${subject}`
+           }),
+           transformResponse: (response: {results: IsaacPodDTO[]; totalResults: number}, meta, arg) => {
+               // Sort news pods in order of id (asc or desc depending on orderDecending), with ones tagged "featured"
+               // placed first
+               return response.results.sort((a, b) => {
+                   const aIsFeatured = a.tags?.includes(FEATURED_NEWS_TAG);
+                   const bIsFeatured = b.tags?.includes(FEATURED_NEWS_TAG);
+                   if (aIsFeatured && !bIsFeatured) return -1;
+                   if (!aIsFeatured && bIsFeatured) return 1;
```
[the remark this answers is not in the document]
Comment (chrisjpurdy) [reply src/app/state/slices/api/index.ts:115]: I don't think it should - featured items should always be ordered first

File: src/app/components/elements/PageFragment.tsx [hunk 2]
```diff
@@ -15,12 +13,8 @@ interface PageFragmentComponentProps {
...
 
 export const PageFragment = ({fragmentId, ifNotFound}: PageFragmentComponentProps) => {
-    const dispatch = useAppDispatch();
-    const fragment = useAppSelector((state: AppState) => state && state.fragments && state.fragments[fragmentId] || null);
-
-    useEffect(() => {
-        dispatch(fetchFragment(fragmentId))
-    }, [dispatch, fragmentId]);
+    const {data: fragment, error} = isaacApi.endpoints.getPageFragment.useQuery(fragmentId);
+    const fragmentOrNotFound = error && 'status' in error && error.status === NOT_FOUND ? NOT_FOUND : fragment;
```
[the remark this answers is not in the document]
Comment (chrisjpurdy) [reply src/app/components/elements/PageFragment.tsx]: Good idea, done!

File: src/app/state/slices/api/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mlt47) [reply src/app/state/slices/api/index.ts:120]: Hmm, yes, I suppose you're right! 👍

**Review by mlt47 — APPROVED** [review]

**Pull request merged**